package/benchmarks/hpobench_nn/_hpobench.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from hpo_benchmarks import HPOBench
+import optuna
+import optunahub
+
+
+_INDEX_SUFFIX = "_index"
+_DIRECTIONS = {
+    "minimize": optuna.study.StudyDirection.MINIMIZE,
+    "maximize": optuna.study.StudyDirection.MAXIMIZE,
+}
+
+
+def _extract_search_space(bench: HPOBench) -> dict[str, optuna.distributions.BaseDistribution]:
```

Comment:
This is a simple question.

The original search space consists of integers and floats, with a limited number of valid values ​​that are not evenly spaced.
https://github.com/automl/HPOBench/blob/master/hpobench/benchmarks/ml/nn_benchmark.py#L36-L52

Is it better to encode parameters into evenly spaced integers (i.e., the current implementation) rather than treating them as floats and rounding them to the nearest valid value when sampling?

Reply:
> The original search space consists of integers and floats, with a limited number of valid values ​​that are not evenly spaced.

They are evenly distributed either as is or in the log space.

> It seems that the original implementation's search space is implemented as a mixture of integers and floats.

Yes and no.
For the tabular benchmark, they use the search space defined in `simple-hpo-bench`.
For the "raw" benchmark where we train the pipeline with a specified configuration, the search space in the URL you provided will be used.